pkg/serverless/metrics/metric_test.go

```diff
@@ -21,17 +22,24 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/aggregator"
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/dogstatsd"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
 )
 
+func TestMain(m *testing.M) {
+	// setting the hostname cache saves about 1s when starting the metric agent
+	cacheKey := cache.BuildAgentKey("hostname")
+	cache.Cache.Set(cacheKey, hostname.Data{}, cache.NoExpiration)
+	os.Exit(m.Run())
+}
+
 func TestStartDoesNotBlock(t *testing.T) {
 	config.DetectFeatures()
 	metricAgent := &ServerlessMetricAgent{}
 	defer metricAgent.Stop()
 	metricAgent.Start(10*time.Second, &MetricConfig{}, &MetricDogStatsD{})
 	assert.NotNil(t, metricAgent.Demux)
 	assert.True(t, metricAgent.IsReady())
-	// allow some time to stop to avoid 'can't listen: listen udp 127.0.0.1:8125: bind: address already in use'
-	time.Sleep(100 * time.Millisecond)
```

Comment:
This wasn't actually doing what we were thinking it was doing. It was included to allow some time to wait after stopping the metric agent. But because the metric agent is stopped with the line `defer metricAgent.Stop()` the sleep was actually happening **before** the metric agent was stopped!